idaes/generic_models/unit_models/distillation/tray_column.py

```diff
@@ -0,0 +1,311 @@
+        self.tray_index = RangeSet(1, self.config.number_of_trays)
+
+        # Add trays
+
+        # TODO:
+        # 1. Add support for multiple feed tray locations
+        # 2. Add support for specifying which trays have side draws
+        self.tray = Tray(self.tray_index,
+                         default={"has_liquid_side_draw":
+                                  self.config.has_liquid_side_draw,
+                                  "has_vapor_side_draw":
+                                  self.config.has_vapor_side_draw,
+                                  "has_heat_transfer":
+                                  self.config.has_heat_transfer,
+                                  "has_pressure_change":
+                                  self.config.has_pressure_change,
+                                  "property_package":
+                                  self.config.property_package,
+                                  "property_package_args":
+                                  self.config.property_package_args},
+                         initialize={self.config.feed_tray_location:
+                                     {"is_feed_tray": True,
+                                      "has_liquid_side_draw":
+                                      self.config.has_liquid_side_draw,
+                                      "has_vapor_side_draw":
+                                      self.config.has_vapor_side_draw,
+                                      "has_heat_transfer":
+                                      self.config.has_heat_transfer,
+                                      "has_pressure_change":
+                                      self.config.has_pressure_change,
+                                      "property_package":
+                                      self.config.property_package,
+                                      "property_package_args":
+                                      self.config.property_package_args}})
+
+        # Add condenser
+        self.condenser = Condenser(
+            default={"condenser_type": self.config.condenser_type,
+                     "temperature_spec":
+                         self.config.condenser_temperature_spec,
+                     "property_package": self.config.property_package,
+                     "property_package_args":
+                     self.config.property_package_args})
+
+        # Add reboiler
+        self.reboiler = Reboiler(
+            default={"has_boilup_ratio": True,
+                     "has_pressure_change": self.config.has_pressure_change,
+                     "property_package": self.config.property_package,
+                     "property_package_args":
+                     self.config.property_package_args})
+
+        # Construct arcs between trays, condenser, and reboiler
+        self._make_arcs()
+        TransformationFactory("network.expand_arcs").apply_to(self)
+
+    def _make_arcs(self):
+        # make arcs
```

Comment:
A comment for future work: One option for avoiding issues with the SD tool would be to write our won constraints for linking trays rather than using Arcs.

A second point - we should use test the Pyomo model reduction tools against this to see if we can eliminate these additional variables and constraints via post-processing. We have a lot of redundant components here, so this would be a really good test case for those tools, and if they work here they will hopefully work for a flowsheet as well.

Reply:
I still would prefer using Pyomo's `Arcs` for this and instead fix the SD tool that looks at the top level. I still need to file the issue with Pyomo. As for the second point, I agree. Something to check.